### PR TITLE
fix(entityframework): Fixed soft delete filtering failing in some cases

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/Data/Masa.Utils.Data.EntityFrameworkCore/Internal/DataFilterState.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Masa.Utils.Data.EntityFrameworkCore.Internal;
+
+internal class DataFilterState
+{
+    public bool IsEnabled { get; set; }
```
Comment (doddgu) [inline src/Data/Masa.Utils.Data.EntityFrameworkCore/Internal/DataFilterState.cs:5]: IsEnabled -> Enabled

Comment (zhenlei520) [reply src/Data/Masa.Utils.Data.EntityFrameworkCore/Internal/DataFilterState.cs]: processed

File: src/Data/Masa.Utils.Data.EntityFrameworkCore/Filters/DataFilter.cs [hunk 2]
```diff
@@ -32,21 +32,29 @@ private DataFilter<TFilter> GetFilter<TFilter>()
...
+    private readonly AsyncLocal<DataFilterState> _filter;
 
     public DataFilter()
     {
-        _filter = new AsyncLocal<bool> { Value = true };
+        _filter = new AsyncLocal<DataFilterState>();
     }
 
-    public bool IsEnabled => _filter.Value;
+    public bool IsEnabled
```
Comment (doddgu) [inline src/Data/Masa.Utils.Data.EntityFrameworkCore/Filters/DataFilter.cs:42]: IsEnabled -> Enabled

Comment (zhenlei520) [reply src/Data/Masa.Utils.Data.EntityFrameworkCore/Filters/DataFilter.cs]: processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**